### DISCUSSION TITLE
Svelte Preprocessor

### DISCUSSION THREAD

**Discussion by TorstenDittmann** [markdoc/markdoc#435] [Show and tell]
I've been exploring Markdoc for work and since we use Svelte I had to check out to make them both work in harmony. Already existing libraries have very limited features of Markdoc implemented and are not really maintained.

Since I guess there might be a need, I'll actually build a proper library now 🙂 

Currently it supports:
- Markdoc support
- Use Svelte components for Nodes (headings, paragraphs, etc) and Tags (custom components)
- Layouts (default or named ones with frontmatter)
- Partials (automatically import all partials from a folder)
- Typescript/JSDoc Support (infer types from the component into the tag)
- Syntax highlighting and autocompletion (experimental, it basically autogenerates a schema for the official markdoc vscode extension)

Looking forward for ideas and improvements regarding developer experience.

Docs: https://svelte-markdoc-preprocess.pages.dev
GitHub: https://github.com/TorstenDittmann/svelte-markdoc-preprocess

PS: The repo is kinda all over the place as of right now and also the docs are in quite early stages, expect bugs and missing stuff

**Comment by mfix-stripe** [comment]
Rad! Thanks for sharing @TorstenDittmann!
